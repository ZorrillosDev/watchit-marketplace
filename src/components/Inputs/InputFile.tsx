// MUI IMPORTS
import {
  Box, Button, InputLabel, styled, ButtonProps,
  InputLabelProps, Typography, BoxProps, TypographyProps
} from '@mui/material'

/* eslint-disable  @typescript-eslint/strict-boolean-expressions */

// ===========================|| INPUT - FILE ||=========================== //

export const InputFileElement = styled('input')({
  display: 'none'
})

export const InputFileIcon = styled(Box)<BoxProps>(({ theme }) => ({
  svg: {
    color: theme.palette.text.secondary,
    width: '2rem',
    height: '2rem'
  }
}))

export const InputFileLabelText = styled(Typography)<TypographyProps>(({ theme }) => ({
  position: 'absolute',
  left: 0,
  top: '-2rem'
}))

export const InputFileLabel = styled(InputLabel, {
  shouldForwardProp: (prop) => prop !== 'isFull'
})<InputLabelProps & { filled: boolean, isFull: boolean }>(({ theme, filled, isFull }) => ({
  border: `2px dashed ${theme.palette.divider}`,
  minHeight: '150px',
  width: '100%',
  padding: filled ? '16px' : '32px 60px',
  position: 'relative',
  borderRadius: '16px',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column',
  overflow: 'visible',
  marginBottom: '1.5rem',
  marginTop: '1.5rem',
  cursor: 'pointer',
  '&:hover': {
    'h5, svg': {
      color: theme.palette.text.primary
    },
    borderColor: theme.palette.text.primary
  },
  '&.Mui-error': {
    'h5, svg': {
      color: theme.palette.error.main
    },
    borderColor: theme.palette.error.main
  },
  '& > .MuiPaper-root': {
    maxWidth: isFull ? '100%' : '16rem'
  }
}))

export const InputFileButton = styled(Button)<ButtonProps>(() => ({
  marginBottom: '-1.5rem',
  transform: 'translateY(2rem)',
  width: '14rem',
  height: '3rem',
  borderRadius: '12px !important',
  boxShadow: '0 5px 9px rgb(0 0 0 / 20%)',
  fontWeight: 600
}))
