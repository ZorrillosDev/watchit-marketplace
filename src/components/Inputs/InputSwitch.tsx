// REACT IMPORTS
import React, { FC, useState } from 'react';

// MUI IMPORTS
import {
  Accordion, AccordionDetails, AccordionSummary, SwitchProps,
  Box, styled, Typography, BoxProps, Switch,
} from '@mui/material';

/* eslint-disable  @typescript-eslint/strict-boolean-expressions */

// ===========================|| INPUTS - SWITCH ||=========================== //

export interface InputSwitchProps {
  label: string | JSX.Element
  subtitle: string | JSX.Element
  content?: JSX.Element
  defaultChecked?: boolean
}

export const InputSwitch: FC<InputSwitchProps> = (props): JSX.Element => {
  const [checked, setChecked] = useState(props.defaultChecked ?? false);

  const handleChange = (event: any): void => {
    setChecked(event.target.checked);
  };

  return (
    <InputSwitchWrapper>
      <Box display='flex' flexDirection='row' alignItems='center' width={1}>
        <Box display='flex' flexDirection='column' flexGrow={1}>
          <Typography variant='body2' color='primary' fontWeight={600}>{props.label}</Typography>
          <Typography variant='body1' color='primary' sx={{ opacity: 0.8 }}>{props.subtitle}</Typography>
        </Box>
        <InputSwitchElement
          checked={checked}
          onChange={handleChange}
          inputProps={{ 'aria-label': 'controlled' }}
        />
      </Box>
      <Accordion expanded={checked} sx={{ m: '0 !important', width: 1, '&:before': { display: 'none' } }}>
        <AccordionSummary sx={{ minHeight: '0px !important', height: 0 }} />
        <AccordionDetails sx={{ p: 0, pt: props.content ? '1rem' : 0 }}>
          {props.content}
        </AccordionDetails>
      </Accordion>
    </InputSwitchWrapper>
  );
};

const InputSwitchWrapper = styled(Box)<BoxProps>(() => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column',
  width: '100%',
}));

const InputSwitchElement = styled(Switch)<SwitchProps>(({ theme }) => ({
  width: 42,
  height: 26,
  padding: 0,
  '& .MuiSwitch-switchBase': {
    padding: 0,
    margin: 2,
    transitionDuration: '300ms',
    '&.Mui-checked': {
      transform: 'translateX(16px)',
      color: '#fff',
      '& + .MuiSwitch-track': {
        backgroundColor: theme.palette.primary.main,
        opacity: 1,
        border: 0,
      },
      '&.Mui-disabled + .MuiSwitch-track': {
        opacity: 0.5,
      },
    },
    '&.Mui-focusVisible .MuiSwitch-thumb': {
      color: theme.palette.primary.main,
      border: '6px solid #fff',
    },
    '&.Mui-disabled .MuiSwitch-thumb': {
      color:
        theme.palette.mode === 'light'
          ? theme.palette.grey[100]
          : theme.palette.grey[600],
    },
    '&.Mui-disabled + .MuiSwitch-track': {
      opacity: theme.palette.mode === 'light' ? 0.7 : 0.3,
    },
  },
  '& .MuiSwitch-thumb': {
    boxSizing: 'border-box',
    width: 22,
    height: 22,
  },
  '& .MuiSwitch-track': {
    borderRadius: 26 / 2,
    backgroundColor: theme.palette.mode === 'light' ? '#E9E9EA' : '#39393D',
    opacity: 1,
    transition: theme.transitions.create(['background-color'], {
      duration: 500,
    }),
  },
}));
