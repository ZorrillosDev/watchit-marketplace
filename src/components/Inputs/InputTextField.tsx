// MUI IMPORTS
import { TextField, styled, TextFieldProps } from '@mui/material'

// ===========================|| INPUT - TEXT - FIELD ||=========================== //

export const InputTextField = styled(TextField)<TextFieldProps>(({ theme }) => ({
  width: '100%',
  'label, input, textarea': {
    color: theme.palette.text.primary,
    fontWeight: 600
  }
}))
