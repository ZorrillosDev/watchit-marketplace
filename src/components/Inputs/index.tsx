// ===========================|| POSTER EXPORTS ||=========================== //

export { InputTextField as default } from '@components/Inputs/InputTextField'
export { InputSwitch } from '@components/Inputs/InputSwitch'
export * from '@components/Inputs/InputFile'
