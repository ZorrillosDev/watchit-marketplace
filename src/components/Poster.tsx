// mui imports
import {
  Avatar,
  AvatarGroup,
  Box, CardHeader,
  Grid,
  styled,
  Typography,
  IconButton, CardMedia,
  CardContent, Card, CardProps, CardHeaderProps, CardContentProps, Menu, MenuItem, Tooltip
} from '@mui/material'
import React, { FC } from 'react'
import { MoreHoriz } from '@mui/icons-material'
import HeartCounter from '@components/HeartCounter'
import TruncatedTypography from '@components/TruncatedTypography'

// ===========================|| POSTER ||=========================== //

export interface PosterProps {
  creator: {
    username: string
    profileUrl: string
  }
  owner: {
    username: string
    profileUrl: string
  }
  posterUrl: string
  price: number
  title: string
  description: string
  rate: number
  isFavorite: boolean
}

const Poster: FC<PosterProps & {showDetails: boolean}> = (props): JSX.Element => {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
  const open = Boolean(anchorEl)

  const handleClick = (event: React.MouseEvent<HTMLElement>): void => {
    setAnchorEl(event.currentTarget)
  }

  const handleClose = (): void => {
    setAnchorEl(null)
  }

  return (
    <PosterWrapper showDetails={props.showDetails}>
      {props.showDetails && (
        <>
          <PosterHeader
            avatar={
              <AvatarGroup max={2}>
                <Tooltip title={`Owner: ${props.owner.username}`}>
                  <Avatar alt={`${props.owner.username} profile image`} src={`${props.owner.profileUrl}`} sx={{ width: 24, height: 24 }} />
                </Tooltip>
                <Tooltip title={`Creator: ${props.creator.username}`}>
                  <Avatar alt={`${props.creator.username} profile image`} src={`${props.creator.profileUrl}`} sx={{ width: 24, height: 24 }} />
                </Tooltip>
              </AvatarGroup>
            }
            action={
              <IconButton aria-label='settings' onClick={handleClick}>
                <MoreHoriz />
              </IconButton>
            }
          />
          <Menu
            anchorEl={anchorEl}
            open={open}
            onClose={handleClose}
            transformOrigin={{
              vertical: 'top',
              horizontal: 'right'
            }}
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'right'
            }}
          >
            <MenuItem onClick={handleClose}>
              Purchase
            </MenuItem>
            <MenuItem onClick={handleClose}>
              Share
            </MenuItem>
            <MenuItem onClick={handleClose}>
              Report
            </MenuItem>
          </Menu>
        </>
      )}
      <PosterMediaContent showDetails={props.showDetails}>
        <CardMedia
          component='img' image={`${props.posterUrl}`} alt={`${props.title}`}
          sx={{ pointerEvents: 'none', userSelect: 'none' }}
        />
      </PosterMediaContent>
      {props.showDetails && (
        <PosterContent>
          <Grid container spacing={1}>
            <Grid item xs={12}>
              <TruncatedTypography variant='body2' color='primary.dark' fontWeight={600} lines={1} sx={{ mt: 0 }}>
                {`${props.title}`}
              </TruncatedTypography>
            </Grid>
            <Grid item xs={6}>
              <Box display='flex' alignItems='center'>
                <Typography variant='body1' display='inline' color='text.primary' fontWeight='bold'>{props.price} ETH</Typography>
              </Box>
            </Grid>
            <Grid item display='flex' alignItems='center' justifyContent='flex-end' xs={6}>
              <HeartCounter count={props.rate} favorite={props.isFavorite} />
            </Grid>
          </Grid>
        </PosterContent>
      )}
    </PosterWrapper>
  )
}

export default Poster

export const PosterWrapper = styled(Card, {
  shouldForwardProp: (prop) => prop !== 'showDetails'
})<CardProps & {showDetails: boolean}>(({ theme, showDetails }) => ({
  position: 'relative',
  backgroundColor: theme.palette.background.paper,
  boxShadow: showDetails ? '0 3px 6px rgba(0,0,0,0.16)' : '',
  cursor: 'pointer',
  width: '100%',
  height: '100%',
  transition: 'all 0.3s ease-in-out',
  '&:hover': {
    transform: 'translateY(-1rem)'
  }
}))

export const PosterHeader = styled(CardHeader)<CardHeaderProps>(() => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  width: '100%',
  padding: '0.25rem 0.5rem'
}))

export const PosterContent = styled(CardContent)<CardContentProps>(() => ({
  padding: '0.5rem !important',
  svg: {
    width: '0.9rem',
    height: '0.9rem'
  }
}))

export const PosterMediaContent = styled(CardContent, {
  shouldForwardProp: (prop) => prop !== 'showDetails'
})<CardContentProps & {showDetails: boolean}>(({ showDetails }) => ({
  padding: '0.5rem !important',
  width: '100%',
  height: '100%',
  maxHeight: showDetails ? 'calc(100% - 7rem)' : 'auto',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  img: {
    height: 'auto',
    maxHeight: '100%',
    maxWidth: '100%',
    borderRadius: '6px',
    width: showDetails ? 'auto !important' : '100%'
  }
}))
