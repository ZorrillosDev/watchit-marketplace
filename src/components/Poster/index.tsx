// ===========================|| POSTER EXPORTS ||=========================== //

export { Poster as default } from '@components/Poster/Poster';
export { PosterWrapper } from '@components/Poster/Poster';
export * from '@components/Poster/PosterFooter';
export * from '@components/Poster/PosterHeader';
export * from '@components/Poster/PosterMedia';
