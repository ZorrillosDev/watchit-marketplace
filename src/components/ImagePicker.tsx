// REACT IMPORTS
import React, { FC } from 'react'

// THIRD PARTY IMPORTS
import { IconUpload } from '@tabler/icons'

// MUI IMPORTS
import { FormControl, Typography } from '@mui/material'

// PROJECT IMPORTS
import {
  InputFileButton, InputFileElement, InputFileIcon,
  InputFileLabel, InputFileLabelText
} from '@components/Inputs'
import { Translation } from '@src/i18n'
import Poster from '@components/Poster'

/* eslint-disable  @typescript-eslint/strict-boolean-expressions */

// ===========================|| INPUT - FILE ||=========================== //

export interface ImagePickerProps {
  id: string
  name: string
  title: string | JSX.Element
  preview: string
  error: boolean
  accept: string
  helpText: string | JSX.Element
  isImageFullWidth?: boolean
  handleChange: (e: any) => void
}

const ImagePicker: FC<ImagePickerProps> = (props): JSX.Element => {
  const handleButtonClick = (e: any): void => {
    e.target.parentElement.click()
  }

  return (
    <FormControl error={props.error} variant='standard' sx={{ width: 1 }}>
      <InputFileLabel htmlFor={props.id} filled={!!props.preview} isFull={!!props.isImageFullWidth}>
        <InputFileLabelText variant='h5' color='text.primary' fontWeight={600}>
          {props.title}
        </InputFileLabelText>
        <InputFileElement
          id={props.id}
          name={props.name}
          aria-describedby={props.id}
          onChange={props.handleChange}
          type='file' accept={props.accept}
        />
        {
            props.preview
              ? (
                <>
                  <Poster image={props.preview} name={`${props.id}_image`} />
                </>
                )
              : (
                <>
                  <InputFileIcon>
                    <IconUpload stroke={1} />
                  </InputFileIcon>
                  <Typography variant='h5' color='text.secondary' width={1} textAlign='center'>
                    {props.helpText}
                  </Typography>
                  <InputFileButton variant='contained' color='primary' onClick={handleButtonClick}>
                    <Translation target='MOVIE_CREATE_FILE_BUTTON' />
                  </InputFileButton>
                </>
                )
        }
      </InputFileLabel>
    </FormControl>
  )
}

export default ImagePicker
