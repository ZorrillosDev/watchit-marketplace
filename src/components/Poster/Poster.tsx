// REACT IMPORTS
import React, { FC } from 'react'

// MUI IMPORTS
import { styled, Card, CardProps } from '@mui/material'

// PROJECT IMPORTS
import { PosterMediaProps, PosterMedia } from '@components/Poster/PosterMedia'

// ===========================|| POSTER ||=========================== //

export const Poster: FC<PosterMediaProps> = (props): JSX.Element => {
  return (
    <PosterWrapper sx={{ border: 'none' }}>
      <PosterMedia {...props} sx={{ maxHeight: 'none', img: { width: '100%' } }} />
    </PosterWrapper>
  )
}

export const PosterWrapper = styled(Card)<CardProps>(() => ({
  position: 'relative',
  backgroundColor: 'transparent',
  border: '1px solid #eee',
  boxShadow: 'none',
  cursor: 'pointer',
  width: '100%',
  height: '100%',
  transition: 'all 0.3s ease-in-out',
  '&:hover': {
    transform: 'translateY(-1rem)'
  }
}))
