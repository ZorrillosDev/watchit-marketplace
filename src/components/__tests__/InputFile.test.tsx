import React from 'react';
import { shallow } from 'enzyme';
import ImagePicker from '@components/ImagePicker';
import Poster from '@components/Poster';
import { InputFileIcon } from '@components/Inputs';

/* eslint-disable  @typescript-eslint/consistent-type-assertions */

describe('<ImagePicker />', () => {
  it('should render', () => {
    const component = shallow(
      <ImagePicker
        error={false} id='hello' title='hello' handleChange={() => {}}
        accept='' helpText='' name='hello' preview='hello'
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('should render movie create poster media', () => {
    const component = shallow(
      <ImagePicker
        error={false} id='world' title='world' handleChange={() => {}}
        accept='' helpText='' name='world' preview='world'
      />,
    );

    expect(component.find(Poster).exists()).toBeTruthy();
    expect(component.find(InputFileIcon).exists()).toBeFalsy();
  });

  it('should render movie create poster media placeholder', () => {
    const component = shallow(
      <ImagePicker
        error={false} id='test' title='test' handleChange={() => {}}
        accept='' helpText='' name='test' preview=''
      />,
    );

    expect(component.find(InputFileIcon).exists()).toBeTruthy();
    expect(component.find(Poster).exists()).toBeFalsy();
  });
});
