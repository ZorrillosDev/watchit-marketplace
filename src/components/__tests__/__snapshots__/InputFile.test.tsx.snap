// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImagePicker /> should render 1`] = `ShallowWrapper {}`;
